aces/Loris
# [Login] Summary Statistics

## Brief summary of changes

- Added a customizable summary of LORIS statistics to the Login page for guest users to see
- A new table was created that is populated with the counts of pinned dataquery queries, and separated by project
- This table is read by the login module for projects with `showSummaryOnLogin` as true in the Project database table.
- If no data is populated, then the summary is not shown on the login page
- Data is populated by the new tool: `tools/update_login_summary_statistics.php`. It must be run for the data in the summary to be updated. Can be added to a project's nightly cron, or at each release
- Some queries are not necessarily possible in the DQT, so in-order to keep it customizable for projects, I added the ability to add SQL files to a folder in project/tools/Login_Summary_Statistics that can be added to the summary list.
- There is a clipboard icon at the bottom right where if you press it, your clipboard will be filled with a csv version of the data
<img width="1170" alt="image" src="https://github.com/user-attachments/assets/02801b16-7d94-48d0-ad2b-ba832d8e4a29" />


#### Testing instructions (if applicable)

1. Run the patch and the script, then open the login page and see that the default queries from `SQL/Login_Summary_Statistics` have appeared.
2. Login as admin, and go to the dataquery module. Create a query that includes the column Project. Run the query then go back to the dataquery homepage and pin the query to the login page. Give it a name, and leave out the 's' for plural (for example, `Male Participant` instead of `Male Participants`).
3. Run the script, then logout and see that the summary appears with a dropdown of all projects. When you select a different project, the count should change.
4. Change a project to `showSummaryOnLogin` = FALSE in the Project SQL table.
5. Re-run the script and note that the project is no longer in the dropdown, and the 'All Projects' count has decreased.
6. Try creating a new query that you only want to show for one project. To do this, create the query, including the Project column, then add a filter for Project = 'Pumpernickel'. Pin the project as before.
8. Run the script, and open the login page. See that the query is not included in the All Projects section, nor in other projects, and is only shown in Pumpernickel.
9. Now, try adding a query such as this to `project/tools/Login_Summary_Statistics` and see that the default queries have disappeared as it is now overridden.
[01_Site.txt](https://github.com/user-attachments/files/18350837/01_Site.txt)


[CCNA OVERRIDE PR ](https://github.com/aces/CCNA/pull/7416)
